src/core/mac/mac_types.cpp

```diff
@@ -404,6 +405,27 @@ bool KeyMaterial::operator==(const KeyMaterial &aOther) const
 #endif
 }
 
+#if OPENTHREAD_CONFIG_WAKEUP_COORDINATOR_ENABLE || OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE
+uint8_t GetWakeupIdLength(WakeupId aWakeupId)
+{
+    uint8_t zeroBytesCount = 0;
+
+    for (int8_t i = sizeof(WakeupId) - 1; i >= 1; --i)
```

Comment:
How about if we use `uint8_t i` instead? 

This way it should belps avoid weird behaior/warnings on some toolhcians when we do things like `((aWakeupId >> (i * kBitsPerByte)) & 0xFF) == 0)` (i.e. using `i` with `>>`)